src/plugins/drawing-tool/components/drawing-layer.test.tsx

```diff
@@ -0,0 +1,324 @@
+import { render, screen } from "@testing-library/react";
+import React from "react";
+import { ToolTileModel } from "../../../models/tools/tool-tile";
+import { DrawingContentModel } from "../model/drawing-content";
+import { DrawingLayerView } from "./drawing-layer";
+import { EllipseDrawingObjectData, ImageDrawingObjectData, LineDrawingObjectData,
+  RectangleDrawingObjectData, VectorDrawingObjectData } from "../model/drawing-objects";
+  import { DrawingToolDeletion, DrawingToolMove } from "../model/drawing-types";
+
+// The starter tile needs to be registered so the ToolTileModel.create
+// knows it is a supported tile type
+import "../drawing-registration";
+
+let content, drawingLayerProps, drawingLayer, drawingObject;
+
+describe("Drawing Layer Components", () => {
+  describe("Freehand Line", () => {
+    const lineData: LineDrawingObjectData = {
+      type: "line",
+      id: "123",
+      x: 10, y: 10,
+      deltaPoints: [{ dx: 1, dy: 1 }, { dx: 2, dy: 2 }],
+      stroke: "#888888",
+      strokeDashArray: "3,3",
+      strokeWidth: 1
+    };
+
+    it("adds a freehand line", () => {
+      content = DrawingContentModel.create({changes:[
+        JSON.stringify({action: "create", data: lineData})
+      ]});
+
+      drawingLayerProps = {
+        model: ToolTileModel.create({content}),
+        onSetCanAcceptDrop: (tileId?: string) => {
+          throw new Error("Function not implemented.");
+        }
+      };
+      render(<DrawingLayerView {...drawingLayerProps} />);
+      drawingLayer = screen.getByTestId("drawing-layer");
+      drawingObject = drawingLayer.firstChild;
+      expect(drawingObject).toMatchSnapshot();
```

Comment:
It looks like this is repeated many times, probably it can be pulled out into a function that just takes a the `content`.